test/celeritas/ext/G4VecgeomConversion.test.cc

```diff
@@ -0,0 +1,183 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020-2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/ext/RootImporter.test.cc
+//---------------------------------------------------------------------------//
+#include "corecel/Types.hh"
+#include "corecel/io/Logger.hh"
+#include "celeritas/ext/VecgeomData.hh"
+#include "celeritas/ext/detail/G4VecgeomConverter.hh"
+#include "celeritas/geo/GeoParams.hh"
+
+#include "Geant4/G4GDMLParser.hh"
+#include "Geant4/G4GeometryManager.hh"
+#include "Geant4/G4PhysicalVolumeStore.hh"
+#include "celeritas_test.hh"
+
+using namespace celeritas;
+template<MemSpace M>
+using PlacedVolumeT = typename detail::VecgeomTraits<M>::PlacedVolume;
+
+//---------------------------------------------------------------------------//
+// TEST HARNESS
+//---------------------------------------------------------------------------//
+/*!
+ * The \e four-levels.gdml file is used to construct a Geant4 model,
+ * which is then converted into a corresponding Vecgeom model.
+ */
+class G4VecgeomConvTest : public ::testing::Test
+{
+  protected:
+    void SetUp() override
+    {
+        CELER_LOG(info) << "G4VecgeomConvTest::SetUp()";
+        // GeantSetup setup = {"four_levels.gdml", GeantSetupOptions{}};
+        G4GDMLParser parser;
+        std::string  filename
+            = ::test::Test::test_data_path("celeritas", "four-levels.gdml");
+        // parser.SetOverlapCheck(true);
+        bool validate_xml = false;
+        parser.Read(filename.c_str(), validate_xml);
```

Comment:
I'm surprised this works with the google test harness since it's reloading the geometry at each test... usually I've been defining static data for setting up geant4 and vecgeom.

Reply:
New testing classes (designed by Seth) solve the issue of rebuilding the geometry after each test.